src/shell/gmenu.js

```diff
@@ -136,7 +136,7 @@ var ListBox = class ListBox extends PopupMenu.PopupMenuSection {
         // We use this instead of close() to avoid touching finalized objects
         } else {
             this.box.opacity = 255;
-            this.box.width = -1;
+            this.box.width = 0;
```

Comment:
Hmm, this seems to work, although it sort of shouldn't :P 

The mistake I made here was thinking `-1` meant "go back to deciding your own allocation", which is what it does in Gtk. I now see `ClutterActor` has different properties for `width`, `natural-width` and `natural-width-set`. I'll merge this since it works, but we should keep an eye on this.

Reply:
I guess `this.box.x_expand` in the constructor is what probably saves it. (And, perhaps, is also why there's no real point in touching the `width` here at all? Setting it may do literally nothing.)

Reply:
Figured it out. I was using the property accessors instead of the property methods, which do take `-1` to reset the preferred size.

Reply:
Oh, sure. That makes sense.

(THAT WAS SARCASM, BECAUSE IT MAKES NO SENSE AT ALL. COULD YOU TELL I WAS BEING SARCASTIC? :wink: )